src/8_logger.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+goog.provide('Logger');
+// jscs:disable validateIndentation
+/**
+ * @constructor
+ */
+
+Logger = function() {
+  /**
+   * @private
+   * @type {string}
+   */
+  this.level_ = 'error';
+  this.levelsOrdered = [ 'verbose', 'info', 'warn', 'error', 'none' ];
+};
+
+/**
+ * Sets the logging level.
+ * @param {string} level - The logging level to set.
+ */
+Logger.prototype.setLevel = function(level) {
+  if (this.levelsOrdered.indexOf(level) !== -1) {
+    this.level_ = level;
+  }
+  else {
+    console.error(`Invalid log level: ${level}`);
+  }
+};
+
+/**
+ * Logs a message to the console if the log level allows it.
+ * @param {string} level - The logging level of the message.
+ * @param {...*} args - The message to log.
+ */
+Logger.prototype.log = function(level) {
+  var args = Array.prototype.slice.call(arguments, 1);
+  if (this.shouldLog(level)) {
+    switch (level) {
+      case 'info':
```

Comment:
Do you need a `verbose` channel? Otherwise `verbose` and `info` appear to be the same.